File: backend/app/schemas/campaign.py
from typing import Optional
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel, ConfigDict
from app.schemas.campaign_output import CampaignOutputResponse


class CampaignBase(BaseModel):
    title: str
    description: Optional[str] = None
    industry: Optional[str] = None
    target_audience: Optional[str] = None
    budget: Optional[str] = None
    goal: Optional[str] = None
    tone: Optional[str] = None


class CampaignCreate(CampaignBase):
    pass


class CampaignUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    industry: Optional[str] = None
    target_audience: Optional[str] = None
    budget: Optional[str] = None
    goal: Optional[str] = None
    tone: Optional[str] = None


class CampaignResponse(CampaignBase):
    id: UUID
    user_id: UUID
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class CampaignDetailResponse(CampaignResponse):
    output: Optional[CampaignOutputResponse] = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/campaign.ts
export interface CampaignOutput {
  id: string
  campaign_id: string
  persona?: any
  buyer_persona?: any
  marketing_strategy?: any
  google_ads?: any[]
  facebook_ads?: any[]
  instagram_ads?: any[]
  linkedin_ads?: any[]
  keywords?: string[]
  hashtags?: string[]
  budget_breakdown?: Record<string, string>
  status: string
  created_at: string
  updated_at: string
}

export interface Campaign {
  id: string
  user_id: string
  title: string
  description?: string
  industry?: string
  target_audience?: string
  budget?: string
  goal?: string
  tone?: string
  created_at: string
  updated_at: string
}

export interface CampaignDetail extends Campaign {
  output?: CampaignOutput
}

export interface CampaignCreateInput {
  title: string
  description: string
  industry?: string
  target_audience?: string
  budget?: string
  goal?: string
  tone?: string
  channels?: string[]
}
